service/controller/v14/resource/endpoint/delete.go

```diff
@@ -62,61 +60,19 @@ func (r *Resource) newDeleteChangeForDeletePatch(ctx context.Context, obj, curre
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-	ips := cutIPs(currentEndpoint.IPs, desiredEndpoint.IPs)
 
-	if currentEndpoint == nil {
-		return nil, nil // Nothing to do.
-	}
-	if desiredEndpoint == nil {
-		return nil, nil // Nothing to do.
-	}
-	if len(ips) > 0 {
-		return nil, nil
-	}
```

Comment:
This made no sense and I think this even caused the endpoint to be deleted. The operator re-created it right away but this was all so complicated. We actually don't want to delete the endpoint in case there are pod IPs. So we just assign the IPs below and that is it.